webpack/assets/javascripts/react_app/components/bookmarks/form/index.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { reduxForm } from 'redux-form';
+import { connect } from 'react-redux';
+import { required, length } from 'redux-form-validators';
+import Form from '../../common/forms/Form';
+import TextField from '../../common/forms/TextField';
+import * as FormActions from '../../../redux/actions/common/forms';
+
+const submit = ({ name, query, publik }, dispatch, props) => {
+  const { submitForm, url, controller } = props;
+  const values = {
+    name,
+    query,
+    public: publik || false,
+    controller,
+  };
+
+  return submitForm({ url, values, item: 'Bookmark' });
+};
+
+class BookmarkForm extends React.Component {
+  render() {
+    const {
+      handleSubmit, submitting, error, onCancel,
+    } = this.props;
+
+    return (
+      <Form
+        onSubmit={handleSubmit(submit)}
+        onCancel={onCancel}
+        disabled={submitting}
+        submitting={submitting}
+        error={error}
+      >
+        <TextField
```

Comment:
I don't see the benefit of having a `TextField` >> `Field` >> `renderField` >> `CommonForm`.
I think it is unnecessary complexity.
Why won't `Forms` use `Field`>>`CommonForm`?

I start seeing the benefit of exporting form fields components from `patternfly-react` so they will replace the `CommonForm`.

This is how it should be done using `react-bootstrap`/`patternfly-react` components:
```js
<Form horizontal>
  <FormGroup controlId="name" disabled={false} validationState="error">
    <Col componentClass={ControlLabel()} sm={3}>
      Name
    </Col>
    <Col sm={9}>
      <FormControl type="text" disabled={false} />
      <HelpBlock>Error here...</HelpBlock>
    </Col>
  </FormGroup>
</Form>
```

http://rawgit.com/sharvit/patternfly-react/storybook/forms/index.html?knob-Validation%20State=error&knob-Show%20Modal=true&knob-Show%20Feedback%20Icon=false&knob-Right%20aligned=false&knob-Show%20Labels=false&knob-Show%20Help=true&knob-Stacked=false&knob-Show%20Loading=false&knob-Disabled=false&knob-Style=default&knob-Label=Danger%20Will%20Robinson%21&selectedKind=Forms&selectedStory=Horizontal%20Form&full=0&down=1&left=1&panelRight=0&downPanel=storybooks%2Fstorybook-addon-knobs

Reply:
> I start seeing the benefit of exporting form fields components from patternfly-react so they will replace the CommonForm.

Once all of the features around validation errors, tool tips etc are in pf, than :+1:

Reply:
:+1: